src/main/java/randoop/output/FailingTestFilter.java

```diff
@@ -0,0 +1,235 @@
+        String line = lineIterator.next();
+        Matcher messageMatcher = FAILURE_MESSAGE_PATTERN.matcher(line);
+        if (messageMatcher.matches()) {
+          totalFailures = Integer.parseInt(messageMatcher.group(1));
+        }
+      }
+      assert lineIterator.hasNext()
+          : "JUnit has non-zero exit status, but we didn't find a failure";
+
+      /*
+       * Then read the rest of the file to find each failure.
+       * The standard runner gives a numbered list with each entry matching the following pattern:
+       */
+      Pattern failureHeaderPattern =
+          Pattern.compile("\\d+\\)\\s+(" + ID_STRING + ")\\(" + classname + "\\)");
+
+      // Split the class text string so that we can match the line number for the assertion with the code.
+      String[] classLines =
+          classString.split(Globals.lineSep); //use same line break as used to write file
+
+      int failureCount = 0;
+      while (lineIterator.hasNext() && failureCount < totalFailures) {
+        String methodName;
+        int lineNumber;
+        String line = lineIterator.next();
+        Matcher failureMatcher = failureHeaderPattern.matcher(line);
+        if (failureMatcher.matches()) { // found the beginning of a failure
+          failureCount++;
+          methodName = failureMatcher.group(1);
+
+          /*
+           * If the method name in the failure message is not a test method, throw an exception.
+           */
+          if (!methodName.matches(GenTests.TEST_METHOD_NAME_PREFIX + "\\d+")) {
+            if (line.contains("initializationError")) {
+              throw new BugInRandoopException(
+                  "Check configuration of test environment: "
+                      + "initialization error of test in flaky-test filter");
+            } else {
+              throw new BugInRandoopException(
+                  "Unexpected failure in flaky-test filter: " + methodName);
+            }
+          }
+
+          /*
+           * Search for the stacktrace entry corresponding to the test method, and capture the line
+           * number.
+           */
+          Pattern linePattern =
+              Pattern.compile(
+                  "\\s+at\\s+"
+                      + classname
+                      + "\\."
+                      + methodName
+                      + "\\("
+                      + classname
+                      + "\\.java:(\\d+)\\)");
+          lineNumber = -1;
+          while (lineIterator.hasNext() && lineNumber < 0) {
+            line = lineIterator.next();
+            Matcher lineMatcher = linePattern.matcher(line);
+            if (lineMatcher.matches()) {
+              lineNumber = Integer.parseInt(lineMatcher.group(1));
+            }
+          }
+          assert lineNumber - 1 < classLines.length;
+          classLines[lineNumber - 1] = "// flaky: " + classLines[lineNumber - 1];
+        }
+      }
+
+      classString = UtilMDE.join(classLines, Globals.lineSep);
```

Comment:
It's inefficient to create a string just for the purpose of passing it to `writeClass`.  Could you create a version of writeClass that takes `List<String>` rather than `String` for the file text?

Even more efficient would be to avoid writing the .java source code to a file at all.  This shows how to create a SimpleJavaFileObject that lets javac be invoked on code stored in a String:
http://docs.oracle.com/javase/7/docs/api/javax/tools/JavaCompiler.html
Wouldn't that be more efficient than writing the .java source code to a file?
And it could save the overhead of starting a new JVM to run each instance of javac.